src/Parsers/findFunctionSecretArguments.cpp

```diff
@@ -0,0 +1,383 @@
+            {
+                /// We couldn't evaluate the argument so we don't know whether it is 'db.table' or just 'db'.
+                /// Hence we can't figure out whether we should skip one argument 'user' or two arguments 'table', 'user'
+                /// before the argument 'password'. So it's safer to wipe two arguments just in case.
+                /// The last argument can be also a `sharding_key`, so we need to check that argument is a literal string
+                /// before wiping it (because the `password` argument is always a literal string).
+                auto res = npos;
+                if (tryGetStringFromArgument(function, arg_num + 2, nullptr, /* allow_identifier= */ false))
+                {
+                    /// Wipe either `password` or `user`.
+                    res = {arg_num + 2, arg_num + 3};
+                }
+                if (tryGetStringFromArgument(function, arg_num + 3, nullptr, /* allow_identifier= */ false))
+                {
+                    /// Wipe either `password` or `sharding_key`.
+                    if (res == npos)
+                        res.first = arg_num + 3;
+                    res.second = arg_num + 4;
+                }
+                return res;
+            }
+
+            /// Skip the current argument (which is either a database name or a qualified table name).
+            ++arg_num;
+            if (database)
+            {
+                /// Skip the 'table' argument if the previous argument was a database name.
+                ++arg_num;
+            }
+        }
+
+        /// Skip username.
+        ++arg_num;
+
+        /// Do our replacement:
+        /// remote('addresses_expr', db.table, 'user', 'password', ...) -> remote('addresses_expr', db.table, 'user', '[HIDDEN]', ...)
+        /// The last argument can be also a `sharding_key`, so we need to check that argument is a literal string
+        /// before wiping it (because the `password` argument is always a literal string).
+        bool can_be_password = tryGetStringFromArgument(function, arg_num, nullptr, /* allow_identifier= */ false);
+        if (can_be_password)
+            return {arg_num, arg_num + 1};
+
+        return npos;
+    }
+
+
+    std::pair<size_t, size_t> findEncryptionFunctionSecretArguments(const ASTFunction & function)
+    {
+        const auto * expr_list = function.arguments->as<ASTExpressionList>();
+        if (!expr_list)
+            return npos; /// return because we don't want to validate query here
+
+        const auto & arguments = expr_list->children;
+
+        /// We replace all arguments after 'mode' with '[HIDDEN]':
+        /// encrypt('mode', 'plaintext', 'key' [, iv, aad]) -> encrypt('mode', '[HIDDEN]')
+        return {1, arguments.size()};
+    }
+
+
+    std::pair<size_t, size_t> findOrdinaryFunctionSecretArguments(const ASTFunction & function)
+    {
+        if (function.name == "mysql")
```

Comment:
we can combine the first 3 conditions, or maybe better to use switch here as all conditions are with function.name, and easy for combine few cases as well.

Reply:
switch can't be used to compare strings

Reply:
I combined the conditions, and in other places too.

Reply:
oh right, sorry my bad